### PR TITLE
#289 hid add proposed solution button for guests

### LINKED ISSUE #289: Frontend - Hide Add Proposed Solution Button for Guests
### Description

Guests aren't able to add new proposed solutions, so they shouldn't see the button for it either.

### Acceptance Criteria

If you're logged in as a guest, the add proposed solutions button shouldn't appear on the change request page

### Proposed Solution

I think this will be in `ProposedSolutionsList.tsx` file or something like that. Use the `useAuth()` hook to get the current user and check their role.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by rzcheung**
## Changes

Added an if-else statement so that if the user is a 'GUEST', the add proposed solution button would not be visible.

## Notes

Had to refer to the string 'GUEST' instead of the role enum because enum would not function correctly

## Test Cases

Tested logging in with each type of user to ensure the button was visible appropriately.

## Screenshots

Screen Shot 2022-10-05 at 9 18 17 PM
Screen Shot 2022-10-05 at 9 18 25 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #289

### CODE REVIEW

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]
I think you'll have to copy your solution into ProposedSolutionsList.tsx too

**Review by Ari-S-123 — APPROVED** [review]

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]
You need to update the failing tests with the new logic too.

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ const ProposedSolutionsList: React.FC<ProposedSolutionsListProps> = ({
 
   return (
     <>
-      {crReviewed === undefined ? (
+      {crReviewed === undefined || auth.user?.role !== 'GUEST' ? (
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx:54]: also tests are failing

**Review by Ari-S-123 — APPROVED** [review]

**Review by anthonybernardi — APPROVED** [review]
nice!

**Pull request merged**